pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -150,115 +179,133 @@ func (computeCfg *computeServiceConfig) getManagedVnetIDs() map[string]struct{}
-			subscriptionIDs := []string{computeCfg.credentials.SubscriptionID}
-			tenantIDs := []string{computeCfg.credentials.TenantID}
-			locations := []string{computeCfg.credentials.region}
-			queryStr, err := getVMsBySubscriptionIDsAndTenantIDsAndLocationsMatchQuery(subscriptionIDs, tenantIDs, locations)
+	var allVirtualMachines []*virtualMachineTable
+	// make cloud api call for each selector.
+	for _, s := range selectorObjects {
+		if len(s.instanceFilters) != 0 {
+			azurePluginLogger().V(1).Info("Fetching vm resources from cloud",
+				"account", computeCfg.accountNamespacedName, "selector", s.namespacedName, "resource-filters", "configured")
+		}
+
+		var subscriptions []*string
+		subscriptions = append(subscriptions, &computeCfg.credentials.SubscriptionID)
+		var virtualMachines []*virtualMachineTable
+		for _, filter := range s.instanceFilters {
+			virtualMachineRows, _, err := getVirtualMachineTable(computeCfg.resourceGraphAPIClient, filter, subscriptions)
 			if err != nil {
-				azurePluginLogger().Error(err, "query string creation failed", "account", computeCfg.account)
-				return nil, false
+				azurePluginLogger().Error(err, "failed to fetch cloud resources",
+					"account", computeCfg.accountNamespacedName, "selector", s.namespacedName)
+				return nil, err
+			}
+			virtualMachines = append(virtualMachines, virtualMachineRows...)
+		}
+		azurePluginLogger().V(1).Info("Vm instances from cloud",
+			"account", computeCfg.accountNamespacedName, "selector", s.namespacedName,
+			"instances", len(virtualMachines))
+
+		// update VM snapshot specific to the current selector(CES).
+		if updateSnapshot {
+			vmIDToInfoMap := make(map[internal.InstanceID]*virtualMachineTable)
+			for _, vm := range virtualMachines {
+				vmIDToInfoMap[internal.InstanceID(strings.ToLower(*vm.ID))] = vm
 			}
-			queries = append(queries, queryStr)
-			return queries, true
+
+			snapshot := computeCfg.resourcesCache.GetSnapshot()
+			selectorSnapshot := snapshot.(*cacheSnapshot).vmSnapshot
+			vmSnapshot := selectorLevelCacheSnapshot{virtualMachines: vmIDToInfoMap}
+			selectorSnapshot[s.namespacedName] = &vmSnapshot
+			computeCfg.resourcesCache.UpdateSnapshot(&cacheSnapshot{selectorSnapshot, snapshot.(*cacheSnapshot).vpcSnapshot})
 		}
-		allFilters = append(allFilters, filters...)
+		allVirtualMachines = append(allVirtualMachines, virtualMachines...)
 	}
-	return allFilters, true
+
+	return allVirtualMachines, nil
 }
 
 func (computeCfg *computeServiceConfig) DoResourceInventory() error {
 	vnets, err := computeCfg.getVpcs()
 	if err != nil {
-		azurePluginLogger().Error(err, "failed to fetch cloud resources", "account", computeCfg.account)
+		azurePluginLogger().Error(err, "failed to fetch cloud resources", "account", computeCfg.accountNamespacedName)
 		return err
 	}
+	vpcPeers := computeCfg.buildMapVpcPeers(vnets)
 
-	virtualMachines, err := computeCfg.getVirtualMachines()
+	virtualMachines, err := computeCfg.getVirtualMachines(true)
```

Comment:
Call it per selector.. And do the map manipulation here and update the snapshot

Reply:
done